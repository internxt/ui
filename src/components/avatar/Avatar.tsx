import DefaultAvatar from './components/DefaultAvatar';
import PictureAvatar from './components/PictureAvatar';

type SIZE_KEYS = 'xxs' | 'xs' | 'sm' | 'base' | 'lg' | 'xl';

const SIZES: Record<SIZE_KEYS, number> = {
  xxs: 28,
  xs: 36,
  sm: 40,
  base: 48,
  lg: 80,
  xl: 128,
};

/**
 * Renders an avatar component which can be either a picture or a default avatar with initials.
 *
 * @param {Object} props - The properties for the Avatar component.
 * @param {string} props.fullName - The full name of the user, used to generate initials if no image is provided.
 * @param {number} [props.diameter=80] - The diameter of the avatar in pixels. Ignored if `size` is provided.
 * @param {SIZE_KEYS} [props.size] - Predefined size for the avatar. If provided, overrides the `diameter`.
 *                                   The associated value in `SIZES` will be used as the diameter. Possible values are:
 *                                    - `'xxs'`: 28px
 *                                    - `'xs'`: 36px
 *                                    - `'sm'`: 40px
 *                                    - `'base'`: 48px
 *                                    - `'lg'`: 80px
 *                                    - `'xl'`: 128px
 * @param {string|null} [props.src] - The URL of the image to display as the avatar. If not provided, initials are shown
 * @param {string} [props.className=''] - Additional CSS classes to apply to the avatar component.
 * @param {Object} [props.style={}] - Additional inline styles to apply to the avatar component.
 * @returns {JSX.Element} The rendered avatar component.
 */

const Avatar = ({
  src,
  diameter = 80,
  size,
  className = '',
  fullName,
  style = {},
}: {
  fullName: string;
  diameter?: number;
  size?: SIZE_KEYS;
  src?: string | null;
  className?: string;
  style?: Record<string, string | number>;
}): JSX.Element => {
  const diameterValue = size ? SIZES[size] : diameter;

  return src ? (
    <PictureAvatar src={src} diameter={diameterValue} className={className} style={style} />
  ) : (
    <DefaultAvatar diameter={diameterValue} className={className} fullName={fullName} />
  );
};

export default Avatar;
