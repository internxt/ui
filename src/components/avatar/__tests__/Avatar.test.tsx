import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import { Avatar } from '../Avatar';

const FULL_NAME = 'My Internxt';
const IMAGE_SRC = 'https://internxt.com/favicon.ico';

describe('Avatar component', () => {
  it('Avatar with full name (first letters) should render correctly', () => {
    const avatarComponent = render(<Avatar diameter={80} fullName={FULL_NAME} />);
    expect(avatarComponent).toMatchSnapshot();
  });

  it('Avatar with avatar (user image profile) should render correctly', () => {
    const avatarComponent = render(<Avatar fullName={FULL_NAME} diameter={80} src={IMAGE_SRC} />);
    expect(avatarComponent).toMatchSnapshot();
  });

  it('XXS Avatar should render correctly', () => {
    const avatarComponent = render(<Avatar fullName={FULL_NAME} size="xxs" src={IMAGE_SRC} />);
    expect(avatarComponent).toMatchSnapshot();
  });
  it('XS Avatar should render correctly', () => {
    const avatarComponent = render(<Avatar fullName={FULL_NAME} size="xs" src={IMAGE_SRC} />);
    expect(avatarComponent).toMatchSnapshot();
  });

  it('SM Avatar should render correctly', () => {
    const avatarComponent = render(<Avatar fullName={FULL_NAME} size="sm" src={IMAGE_SRC} />);
    expect(avatarComponent).toMatchSnapshot();
  });

  it('Base Avatar should render correctly', () => {
    const avatarComponent = render(<Avatar fullName={FULL_NAME} size="base" src={IMAGE_SRC} />);
    expect(avatarComponent).toMatchSnapshot();
  });
  it('LG Avatar should render correctly', () => {
    const avatarComponent = render(<Avatar fullName={FULL_NAME} size="lg" src={IMAGE_SRC} />);
    expect(avatarComponent).toMatchSnapshot();
  });
  it('XL Avatar should render correctly', () => {
    const avatarComponent = render(<Avatar fullName={FULL_NAME} size="xl" src={IMAGE_SRC} />);
    expect(avatarComponent).toMatchSnapshot();
  });
});
