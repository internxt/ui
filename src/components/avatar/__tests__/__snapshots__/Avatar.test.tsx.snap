// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar component > Avatar with avatar (user image profile) should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        class=" shrink-0 select-none rounded-full object-cover"
        draggable="false"
        src="https://internxt.com/favicon.ico"
        style="width: 80px; height: 80px;"
      />
    </div>
  </body>,
  "container": <div>
    <img
      class=" shrink-0 select-none rounded-full object-cover"
      draggable="false"
      src="https://internxt.com/favicon.ico"
      style="width: 80px; height: 80px;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar component > Avatar with full name (first letters) should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" flex shrink-0 select-none items-center justify-center rounded-full bg-primary/20 font-medium text-primary dark:bg-primary/75 dark:text-white"
        style="width: 80px; height: 80px; font-size: 38.095238095238095px;"
      >
        <p>
          MI
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" flex shrink-0 select-none items-center justify-center rounded-full bg-primary/20 font-medium text-primary dark:bg-primary/75 dark:text-white"
      style="width: 80px; height: 80px; font-size: 38.095238095238095px;"
    >
      <p>
        MI
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar component > Base Avatar should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        class=" shrink-0 select-none rounded-full object-cover"
        draggable="false"
        src="https://internxt.com/favicon.ico"
        style="width: 48px; height: 48px;"
      />
    </div>
  </body>,
  "container": <div>
    <img
      class=" shrink-0 select-none rounded-full object-cover"
      draggable="false"
      src="https://internxt.com/favicon.ico"
      style="width: 48px; height: 48px;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar component > LG Avatar should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        class=" shrink-0 select-none rounded-full object-cover"
        draggable="false"
        src="https://internxt.com/favicon.ico"
        style="width: 80px; height: 80px;"
      />
    </div>
  </body>,
  "container": <div>
    <img
      class=" shrink-0 select-none rounded-full object-cover"
      draggable="false"
      src="https://internxt.com/favicon.ico"
      style="width: 80px; height: 80px;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar component > SM Avatar should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        class=" shrink-0 select-none rounded-full object-cover"
        draggable="false"
        src="https://internxt.com/favicon.ico"
        style="width: 40px; height: 40px;"
      />
    </div>
  </body>,
  "container": <div>
    <img
      class=" shrink-0 select-none rounded-full object-cover"
      draggable="false"
      src="https://internxt.com/favicon.ico"
      style="width: 40px; height: 40px;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar component > XL Avatar should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        class=" shrink-0 select-none rounded-full object-cover"
        draggable="false"
        src="https://internxt.com/favicon.ico"
        style="width: 128px; height: 128px;"
      />
    </div>
  </body>,
  "container": <div>
    <img
      class=" shrink-0 select-none rounded-full object-cover"
      draggable="false"
      src="https://internxt.com/favicon.ico"
      style="width: 128px; height: 128px;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar component > XS Avatar should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        class=" shrink-0 select-none rounded-full object-cover"
        draggable="false"
        src="https://internxt.com/favicon.ico"
        style="width: 36px; height: 36px;"
      />
    </div>
  </body>,
  "container": <div>
    <img
      class=" shrink-0 select-none rounded-full object-cover"
      draggable="false"
      src="https://internxt.com/favicon.ico"
      style="width: 36px; height: 36px;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar component > XXS Avatar should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        class=" shrink-0 select-none rounded-full object-cover"
        draggable="false"
        src="https://internxt.com/favicon.ico"
        style="width: 28px; height: 28px;"
      />
    </div>
  </body>,
  "container": <div>
    <img
      class=" shrink-0 select-none rounded-full object-cover"
      draggable="false"
      src="https://internxt.com/favicon.ico"
      style="width: 28px; height: 28px;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
